### DISCUSSION TITLE
Rotate images by exif info

### DISCUSSION THREAD

**Discussion by gthrepwood** [thumbsup/thumbsup#315] [General]
Maybe it's just me, but is there a way to rotate images by exif rotateion info in the album? Or is that depends on the theme? Thanks!

**Comment by rprieto** [comment]
Hi, what do you expect exactly?

Today, Thumbsup strips all EXIF data in the resized versions, but rotates the images according to the EXIF tag so they're the right way up.

**Comment by amandal1810** [comment]
Thumbsup is able to set the correct orientation for images pulled off the internet. But when using photos from a camera, the orientation is not respected. I am using Windows 10.

I checked the `Orientation` value for both the images using exiftool:
 - for the image off the internet: the `Orientation` property does not exist
 - for the image from a camera: the value for `Orientation` property is `Rotate 270 CW`

**Reply by rprieto** [reply]
Hi. Is this related to https://github.com/thumbsup/thumbsup/issues/317 ? It looks like it could be a recent change in GraphicsMagick, that we have to cater for internally.

**Reply by amandal1810** [reply]
Hey, yes, it is indeed related! Thanks. FWIW, I am using a slightly older version of GraphicsMagick (1.3.35 to be precise) since newer binaries are unavailable for Windows.
BTW, I did try using the `--gm-args strip` parameter, which fixes the orientation of the large images but the thumbnail image gets weirdly stretched out (in the correct orientation).
